rails-snow-queen/app/assets/javascripts/quote.js

```diff
@@ -0,0 +1,63 @@
+class Quote {
+  constructor(config) {
+    this.errors = null;
+    this.onOpen = config.onOpen;
+    
+    this.handleErrors = this.handleErrors.bind(this);
+    this.handleOpenModal = this.handleOpenModal.bind(this);
+    this.addListeners();
+  }
+  
+  addListeners() {
+    $("#submitQuoteModal").on("show.bs.modal", this.handleOpenModal);
+    document.getElementById("new_quote").addEventListener("ajax:error", this.handleErrors);
+    document.getElementById("new_quote").addEventListener("ajax:success", this.handleSuccess);
```

Comment:
No point looking up the node multiple times. Save it in a variable.
```javascript
const form = document.getElementById("new_quote");
form.addEventListener(...);
form.addEventListener(...);
```

Reply:
done